### ISSUE TITLE
Integration test failures

### ISSUE THREAD

**Issue by fooman** [graycoreio/magento2-graphql-introspection-cache#5]
Installed this into a 2.4.5 Community site and got a bunch of integration test failures coming from core and other GraphQl modules. 

```
1) Magento\GraphQlCache\Controller\Catalog\CategoriesWithProductsCacheTest::testRequestCacheTagsForCategoryWithProducts
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'cat_c'
-    1 => 'cat_c_333'
-    2 => 'cat_p'
-    3 => 'cat_p_333'
-    4 => 'FPC'
+    0 => 'FPC'
 )
 ```
 I do not think removal of cache tags would be expected for those queries. 
 
 There are few more failures coming from the tests for Magento_GraphQlCache.

And also a few more along the lines of the below:

```
1) Magento\GraphQl\GraphQlIntrospectionTest::testIntrospectionQuery
PHPUnit\Framework\Exception: Warning: Undefined array key "__schema" in /__w/store.***_m2/store.***_m2/dev/tests/integration/testsuite/Magento/GraphQl/GraphQlIntrospectionTest.php:82.
```

**Comment by damienwebdev** [comment]
@fooman are you able to submit a PR with just an example failing test? 

@danslo I was looking at the `di.xml` and was wondering if it should be in the `graphql` scope. Maybe we're accidentally hitting a bad scope?

**Comment by danslo** [comment]
Hmmm interesting, Magento 2.4.5 installed with this module:
```
www-data@magento24-php-fpm:/var/www/html$ cat composer.json | grep product-community
        "magento/product-community-edition": "2.4.5",
```
```
www-data@magento24-php-fpm:/var/www/html$ cat app/etc/config.php | grep GraphQlIntrospection
        'Graycore_GraphQlIntrospectionCache' => 1,
```

GraphQlIntrospectionTest:
```
www-data@magento24-php-fpm:/var/www/html$ ./vendor/bin/phpunit -c $(pwd)/dev/tests/integration/phpunit.xml.dist dev/tests/integration/testsuite/Magento/GraphQl/GraphQlIntrospectionTest.php
PHPUnit 9.5.23 #StandWithUkraine

Warning:       Using a custom test suite loader is deprecated

...                                                                                                                                                                                                                                                                                                              3 / 3 (100%)

Time: 00:01.441, Memory: 99.50 MB

OK (3 tests, 5 assertions)

=== Memory Usage System Stats ===
Memory usage (OS):      135.80M (142.20% of 95.50M reported by PHP)
Estimated memory leak:  40.30M (29.68% of used memory)
```

CategoriesWithProductsCacheTest:
```
www-data@magento24-php-fpm:/var/www/html$ ./vendor/bin/phpunit -c $(pwd)/dev/tests/integration/phpunit.xml.dist dev/tests/integration/testsuite/Magento/GraphQlCache/Controller/Catalog/CategoriesWithProductsCacheTest.php
PHPUnit 9.5.23 #StandWithUkraine

Warning:       Using a custom test suite loader is deprecated

.                                                                                                                                                                                                                                                                                                                1 / 1 (100%)

Time: 00:01.869, Memory: 133.00 MB

OK (1 test, 2 assertions)

=== Memory Usage System Stats ===
Memory usage (OS):      181.42M (136.40% of 133.00M reported by PHP)
Estimated memory leak:  48.42M (26.69% of used memory)
```

I tried analyzing how we can possibly be removing cache tags, but we only call [addCacheTags](https://github.com/graycoreio/magento2-graphql-introspection-cache/blob/main/Plugin/CachePlugin.php#L47-L49) which just [merges](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/GraphQlCache/Model/CacheableQuery.php#L43) cache tags.

@fooman Is there an easy way for you to make this reproducible in a clean install?

**Comment by danslo** [comment]
Nevermind, I know what's going on. Working on a fix.

**Comment by damienwebdev** [comment]
Fixed by #6

**Comment by damienwebdev** [comment]
Released as [v0.1.2](https://github.com/graycoreio/magento2-graphql-introspection-cache/releases/tag/v0.1.2).
